enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,372 @@
+  and `cno-mtu-migration-workers` with `TARGET_MTU` unset.
+* MCO performs a rolling reboot of the nodes. Nodes reboot with MTU set to
+  `1400` for OVN-K interfaces and `1500` for host interfaces. The MTU decrease
+  will not have impact since the cluster was already effectively using that MTU
+  due to the routes MTU being previously set.
+
+### MTU increase example
+
+Current MTU: 1400 Target MTU: 9000
+
+* Administrator updates the CNO operator configuration, changing `mtu` from
+  `1400` to `9000` and annotating it with `mtu-migration` set to `9100`.
+* CNO renders and applies the OVN-K configuration with `mtu` set to `9000`
+  (changed from `1400`) and `routable-mtu` set to `1400` (previously unset). It
+  also renders and applies two MachineConfig objects
+  `cno-mtu-migration-masters` and `cno-mtu-migration-workers` with
+  `TARGET_MTU` set to `9100`.
+* MCO performs a rolling reboot of the nodes. Nodes reboot with MTU set to
+  `9000` for OVN-K interfaces and `9100` for host interfaces, and routes MTU set
+  to `1400` for pod traffic and the previous host MTU for host traffic. As nodes
+  restart they will preserve connectivity because even though the interfaces MTU
+  increased the effective MTU in use did not change.
+* CNO renders and applies the OVN-K configuration with `mtu` set to `9000`
+  (unchanged) and `routable-mtu` unset (changed form `1400`). It also renders
+  and applies two MachineConfig objects `cno-mtu-migration-masters` and
+  `cno-mtu-migration-workers` with `TARGET_MTU` unset.
+* MCO performs a rolling reboot of the nodes. Nodes reboot interfaces MTU
+  unchanged and no routes MTU set effectively increasing the MTU in use but with
+  no impact since the interfaces MTU are already set to appropriate values.
+
+### User Stories
+
+#### As an administrator, I want to change the cluster MTU after deployment.
+
+### Implementation Details/Notes/Constraints
+
+## Design Details
+
+### Open Questions
+
+* The proposed procedure is completely automated. Should we consider making some
+  of the steps manual as an option? Perhaps indicated through an alternate
+  annotation `manual-mtu-migration` that would have the effect of rendering the
+  MachineConfig objects without a machine pool label. The administrator could
+  then add the label to manually trigger the reboots. We could also offer the
+  option to set which MachineConfig pools to work with, to provide more control
+  over the reboot process. The counterpart is that if reboots are manual,
+  unintended reboots unrelated to this procedure before the label is applied to
+  the Machine configs might lead to a degraded system.
+* We might not be aware at any given time of br-ex link scoped routes if these
+  are dynamically assigned through IPv6 RA. NetworkManager dispatcher does not
+  seem to trigger an event for route modifications resulting from IPv6 RA.
+* Should the host MTU change outlive OVN-K? For example, if migrating back to
```

Comment:
if we migrate back to openshift-sdn, br-ex and the associated connections will be deleted, so whatever MTU the host interfaces connections were configured with will be used. Right?

Reply:
Yes, that would be a rollback to the old MTU and I guess that's what I am asking, would that be desirable or maybe not?

Reply:
hmm I noticed the title of this talks about MTU change generically which makes me thing openshift-sdn is included, but the details of the proposal talk about ovnk changes. Are we trying to support this for openshift-sdn and ovnk or ovnk only?

Reply:
the epic says openshift-sdn as well. We met offline and came up with a plan to make it work for openshift-sdn as well.